src/components/ModalOverlay/ModalOverlay.tsx

```diff
@@ -1,22 +1,39 @@
-import React from 'react';
+import React, { useRef, useEffect } from 'react';
 import { Modal, ModalProps } from '../Modal';
 import { OverlayProps, Overlay } from '../Overlay';
 import classNames from 'classnames';
 
 import './ModalOverlay.scss';
+import { createPortal } from 'react-dom';
 
-export type ModalOverlayProps = ModalProps & OverlayProps;
+export type ModalOverlayProps = ModalProps &
+  OverlayProps & { overlayClassName?: string; contentClassName?: string };
 
 export const ModalOverlay: React.FC<ModalOverlayProps> = ({
   children,
   className,
+  contentClassName,
   onClose,
-  show,
+  overlayClassName,
+  show = true,
   ...rest
-}) => (
-  <Overlay className={classNames('ModalOverlay', className)} show={show} {...rest}>
-    <Modal className='ModalOverlay__modal' onClose={onClose}>
-      <div className='ModalOverlay__content'>{children}</div>
-    </Modal>
-  </Overlay>
-);
+}) => {
+  const el = useRef(document.createElement('div'));
+
+  useEffect(() => {
```

Comment:
why is this only here: 

`<Overlay />` is setting an overlay over the parent element while highlighting its `children`
`<Modal />` is rendering a `fixed` element

`<ModalOverlay />` has to set an overlay over the whole doc while rendering a `<Modal />`